app/controllers/carts_controller.rb

```diff
@@ -7,4 +8,19 @@ def show
     @show_comments = true
   end
 
+  def index
+    @role = 'requester'
+    my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
+    if my_carts.empty?
+      @role = 'approver'
+      my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
+    end
```

Comment:
Well, I agree that there's not likely to be overlap. There are (minor)
differences between how carts are represented depending on whether one is
an approver or a requester.
 We could simplify the query by cheating and just picking the role of the
first cart.
I believe that it's useful to distinguish between open and closed carts.

On Fri, Jan 2, 2015 at 12:16 PM, Aidan Feldman <notifications@github.com>
wrote:

> In app/controllers/carts_controller.rb
> <https://github.com/18F/C2/pull/131#discussion-diff-22421413>:
>
> > @@ -7,4 +8,19 @@ def show
> >      @show_comments = true
> >    end
> >
> > +  def index
> > +    @role = 'requester'
> > +    my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
> > +    if my_carts.empty?
> > +      @role = 'approver'
> > +      my_carts = Cart.joins(:approvals).where(:approvals => {:role => @role, :user_id => @current_user[:id]})
> > +    end
>
> This seems over-complicated code and UX-wise... can we just do this?
>
> my_carts = current_user.carts.order('created_at DESC')
>
> We can do a fancier query
> <http://stackoverflow.com/questions/6332043/sql-order-by-multiple-values-in-specific-order>
> if we want to show carts where the user is a requester before those where
> they're an approver, but
>
>    - I don't think there will be a lot of overlap between requesters and
>    approvers, at least for WHSC
>    - My instinct is that the user would want to see all carts they're
>    involved with in one list anyway
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/C2/pull/131/files#r22421413>.
>